src/cargo/sources/git/oxide.rs

```diff
@@ -0,0 +1,315 @@
+            .config_overrides(config_overrides)
+    })
+}
+
+/// Convert `git` related cargo configuration into the respective `git` configuration which can be
+/// used when opening new repositories.
+pub fn cargo_config_to_gitoxide_overrides(config: &Config) -> CargoResult<Vec<BString>> {
+    use gix::config::tree::{gitoxide, Core, Http, Key};
+    let timeout = HttpTimeout::new(config)?;
+    let http = config.http_config()?;
+
+    let mut values = vec![
+        gitoxide::Http::CONNECT_TIMEOUT.validated_assignment_fmt(&timeout.dur.as_millis())?,
+        Http::LOW_SPEED_LIMIT.validated_assignment_fmt(&timeout.low_speed_limit)?,
+        Http::LOW_SPEED_TIME.validated_assignment_fmt(&timeout.dur.as_secs())?,
+        // Assure we are not depending on committer information when updating refs after cloning.
+        Core::LOG_ALL_REF_UPDATES.validated_assignment_fmt(&false)?,
+    ];
+    if let Some(proxy) = &http.proxy {
+        values.push(Http::PROXY.validated_assignment_fmt(proxy)?);
+    }
+    if let Some(check_revoke) = http.check_revoke {
+        values.push(Http::SCHANNEL_CHECK_REVOKE.validated_assignment_fmt(&check_revoke)?);
+    }
+    if let Some(cainfo) = &http.cainfo {
+        values.push(
+            Http::SSL_CA_INFO.validated_assignment_fmt(&cainfo.resolve_path(config).display())?,
+        );
+    }
+    if let Some(user_agent) = &http.user_agent {
+        values.push(Http::USER_AGENT.validated_assignment_fmt(user_agent)?);
+    }
+    if let Some(ssl_version) = &http.ssl_version {
+        use crate::util::config::SslVersionConfig;
+        match ssl_version {
+            SslVersionConfig::Single(version) => {
+                values.push(Http::SSL_VERSION.validated_assignment_fmt(&version)?);
+            }
+            SslVersionConfig::Range(range) => {
+                values.push(
+                    gitoxide::Http::SSL_VERSION_MIN
+                        .validated_assignment_fmt(&range.min.as_deref().unwrap_or("default"))?,
+                );
+                values.push(
+                    gitoxide::Http::SSL_VERSION_MAX
+                        .validated_assignment_fmt(&range.max.as_deref().unwrap_or("default"))?,
+                );
+            }
+        }
+    }
+    if let Some(debug) = http.debug {
+        values.push(gitoxide::Http::VERBOSE.validated_assignment_fmt(&debug)?);
+    }
+    if let Some(multiplexing) = http.multiplexing {
+        let http_version = multiplexing.then(|| "HTTP/2").unwrap_or("HTTP/1.1");
+        // Note: the HTTP version itself should automatically make multiplexing available.
+        //       It's unclear at this time if it does affect performance of fetches measurably.
+        //       In `curl`, it appears special support via a `Multi` handle is needed which also doesn't
+        //       exist in `gitoxide` - each fetch is probably its own connection that way.
+        values.push(Http::VERSION.validated_assignment_fmt(&http_version)?);
+    }
+
```

Comment:
Should we set an empty `accept-encoding = ""` here as well?

Ref:

- https://github.com/rust-lang/cargo/pull/11292/commits/7c89237104b9ea35ae262b1ed9e43575a1fbd01a
- https://curl.se/libcurl/c/CURLOPT_ACCEPT_ENCODING.html

Reply:
`git` itself also doesn't set it and I would take this as indicator that it's not worth doing, a sentiment I'd also arrive from when looking at the protocol.

Requests `gitoxide` does right now don't set the "accept-encoding" header at all, and comparing this response with one that I manufactured which sets this header shows no difference. This seems to indicate that at least GitHub also doesn't do anything about it.

<img width="722" alt="Screenshot 2023-02-17 at 16 23 57" src="https://user-images.githubusercontent.com/63622/219695147-47c2c23f-c897-4895-8bf2-854d7e1f91a9.png">
<img width="712" alt="Screenshot 2023-02-17 at 16 23 47" src="https://user-images.githubusercontent.com/63622/219695159-7d9803de-f6ba-4309-bfe5-0b434946c216.png">

If you think it should be done for other reasons, I can make this setting available in `gitoxide` though, just let me know.

Reply:
Fair enough. Git has its own protocol and packfile format. I think it make sense not setting `accept-encoding`. Thank you for the investigation!